eval.go

```diff
@@ -98,6 +99,9 @@ func (d *indexData) simplify(in query.Q) query.Q {
 			})
 		case *query.Language:
 			_, has := d.metaData.LanguageMap[r.Language]
+			if !has {
+				_, has = enry_data.ExtensionsByLanguage[r.Language]
+			}
```

Comment:
I don't understand this check. `simplify` is checking if we have the language in the shard. All you should need to check is LanguageMap since that is populated at index time. Checking a static list is unrelated to the content in the shard. Is this some left over testing code?

Reply:
Before, `lang:` simplifies itself to a null match if the language map is entirely missing the specified language.

Because only the ambiguous file identifications are stored in LanguageMap, this is no longer applicable, and we have to fallback to file extension scanning.

Reply:
It sounds like we still want to simplify to null / false? IE don't you need to remove this now that LanguageMap contains all valid languages according to go-enry.

Edit: ok I think I understand. A file may be multiple possible languages. Something here is still a bit fishy to me. I would suspect at index time we need to somehow indicate this is possible or store all possible language types for a file.

As this is implemented, if you ask for `lang:go` and the shard has no go files, we will still search it since `lang:go` is a valid language according to enry_data.